scripts/lint-style.py

```diff
@@ -160,6 +161,40 @@ def line_endings_check(lines, path):
         newlines.append((line_nr, line))
     return errors, newlines
 
+def four_spaces_in_second_line(lines, path):
+    # TODO: also fix the space for all lines before ":=", right now we only fix the line after
+    # the first line break
+    errors = []
+    # We never alter the first line, as it does not occur as next_line in the iteration over the
+    # zipped lines below, hence we add it here
+    newlines = [lines[0]]
+    annotated_lines = list(annotate_comments(lines))
+    for (_, line, is_comment), (next_line_nr, next_line, _) in zip(annotated_lines,
+                                                                   annotated_lines[1:]):
+        # Check if the current line matches "(lemma|theorem) .* :"
+        new_next_line = next_line
+        if (not is_comment) and re.search(r"^(protected )?(def|lemma|theorem) (?!.*:=).*(where)?$",
```

Comment:
```suggestion
        if (not is_comment) and re.search(r"^(protected )?(def|lemma|theorem|instance) (?!.*:=).*(where)?$",
```

Reply:
This also presumably won't match `@[simp] lemma`?

Reply:
Indeed, but we could be here all day trying to reproduce the grammar of lean declarations in python.
Like `irreducible_def`, `private`, `noncomputable`, etc. etc.
I don't think there is anything wrong with adding attributes here too, but after a point we really need to either do these things in lean itself (I assume once someone makes one such linter making the rest will be easy), or use something like tree-sitter at least so we aren't reimplementing this all the time.